### PR TITLE
[docs] add some missing partition mappings to the api docs

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
A couple partition mappings were missing from the API docs - this adds them!

## How I Tested These Changes
looked at the docs site locally

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15205** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15205" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15205?utm_source=stack-comment).

File: docs/sphinx/sections/api/apidocs/partitions.rst [hunk 1]
```diff
@@ -46,6 +46,13 @@ Partition Mapping
 
 .. autoclass:: StaticPartitionMapping
 
+.. autoclass:: SpecificPartitionsPartitionMapping
```
Comment (jamiedemaria) [inline docs/sphinx/sections/api/apidocs/partitions.rst:49]: cc @clairelin135  is there an order of these partition mappings that would be helpful? like to any of them build on each other?

**Review by erinkcochran87 — COMMENTED** [review]
Looks good to me - I'll leave it to Claire to approve since there's an open question to her.

File: docs/sphinx/sections/api/apidocs/partitions.rst (hunk 1 above)
Comment (clairelin135) [reply docs/sphinx/sections/api/apidocs/partitions.rst:49]: The MultiPartitionMapping and MultiToSingleDimensionPartitionMapping do build on other ones, I think it's good that they're listed the last

**Review by clairelin135 — APPROVED** [review]

**Pull request merged**